### PR TITLE
Azure wal

### PR DESCRIPTION
**Pull request by ssinger**
Add support for sending wal via wal-g to azure storage.
This addresses issue  #383

### CODE REVIEW

**Review by CyberDem0n — COMMENTED** [review]
Hi @ssinger, thank you for the PR,
please fix python formatting issues: https://github.com/zalando/spilo/pull/514/checks?check_run_id=1418286759#step:7:6

File: ENVIRONMENT.rst [hunk 1]
```diff
@@ -65,6 +65,9 @@ Environment Configuration Settings
 - **SWIFT_PROJECT_DOMAIN_NAME**:
 - **SWIFT_PROJECT_DOMAIN_ID**:
 - **WALE_SWIFT_PREFIX**: (optional) the full path to the backup location on the Swift Storage in the format swift://bucket-name/very/long/path. If not specified Spilo will generate it from WAL_SWIFT_BUCKET.
+- **WALG_AZ_PREFIX**: (optional) the azure prefix to store WAL backups at in the format azure://test-container/walg-folder.
+- **AZURE_STORAGE_ACCOUNT**:(optional) the azure storage account to use for WAL backups.
+- **AZURE_STORAGE_ACCONT_ACCESS_KEY**:(optional) the access key for the azure storage account used for WAL backups.
```
Comment (carlreid) [inline ENVIRONMENT.rst:70]: There's a typo here:
```suggestion
- **AZURE_STORAGE_ACCOUNT_ACCESS_KEY**:(optional) the access key for the azure storage account used for WAL backups.
```

But I see later in the code the mention of `AZURE_STORAGE_ACCESS_KEY`, so I assume the environment variable should be this instead?

File: postgres-appliance/scripts/configure_spilo.py [hunk 2]
```diff
@@ -423,7 +422,7 @@ def set_extended_wale_placeholders(placeholders, prefix):
 
 def set_walg_placeholders(placeholders, prefix=''):
     walg_supported = any(placeholders.get(prefix + n) for n in AUTO_ENABLE_WALG_RESTORE +
-                         ('WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX'))
+                         ('WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX', 'WALG_AZ_PREFIX'))
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/configure_spilo.py:425]: This change is not required, `AUTO_ENABLE_WALG_RESTORE` already contains `WALG_AZ_PREFIX`.

File: postgres-appliance/scripts/configure_spilo.py [hunk 3]
```diff
@@ -563,7 +562,7 @@ def get_placeholders(provider):
 
     placeholders['USE_WALE'] = any(placeholders.get(n) for n in AUTO_ENABLE_WALG_RESTORE +
                                    ('WAL_SWIFT_BUCKET', 'WALE_SWIFT_PREFIX', 'WAL_GCS_BUCKET',
-                                    'WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX'))
+                                    'WAL_GS_BUCKET', 'WALE_GS_PREFIX', 'WALG_GS_PREFIX', 'WALG_AZ_PREFIX'))
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/configure_spilo.py:565]: This change is not required, `AUTO_ENABLE_WALG_RESTORE` already contains `WALG_AZ_PREFIX`.

File: postgres-appliance/scripts/configure_spilo.py [hunk 4]
```diff
@@ -699,8 +699,9 @@ def write_wale_environment(placeholders, prefix, overwrite):
 
     wale = defaultdict(lambda: '')
     for name in ['WALE_ENV_DIR', 'SCOPE', 'WAL_BUCKET_SCOPE_PREFIX', 'WAL_BUCKET_SCOPE_SUFFIX',
-                 'WAL_S3_BUCKET', 'WAL_GCS_BUCKET', 'WAL_GS_BUCKET', 'WAL_SWIFT_BUCKET', 'BACKUP_NUM_TO_RETAIN'] +\
-            s3_names + swift_names + gs_names + walg_names:
+                 'WAL_S3_BUCKET', 'WAL_GCS_BUCKET', 'WAL_GS_BUCKET', 'WAL_SWIFT_BUCKET', 'BACKUP_NUM_TO_RETAIN',
+                 'WALG_AZ_PREFIX'] +\
+            s3_names + swift_names + gs_names + walg_names + azure_names:
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/configure_spilo.py:704]: Lets merge these two lines, so it look nicely.
```suggestion
                 'WALG_AZ_PREFIX'] + s3_names + swift_names + gs_names + walg_names + azure_names:
```

File: postgres-appliance/scripts/configure_spilo.py [hunk 5]
```diff
@@ -753,7 +756,6 @@ def write_wale_environment(placeholders, prefix, overwrite):
     # Set WALG_*_PREFIX for future compatibility
     if store_type in ('S3', 'GS') and not wale.get(write_envdir_names[1]):
         wale[write_envdir_names[1]] = wale[prefix_env_name]
-
```
Comment (CyberDem0n) [inline postgres-appliance/scripts/configure_spilo.py:756]: Seems like unrelated change.

**Review by CyberDem0n — CHANGES_REQUESTED** [review]
Since currently Azure is not enabled for wal-e it would make sense to automatically set `USE_WALG_BACKUP` to `true` if `WALG_AZ_PREFIX` is defined:

```diff
     if any(placeholders.get(n) for n in AUTO_ENABLE_WALG_RESTORE):
         placeholders.setdefault('USE_WALG_RESTORE', 'true')
+    if placeholders.get('WALG_AZ_PREFIX'):
+        placeholders.setdefault('USE_WALG_BACKUP', 'true')
     set_walg_placeholders(placeholders)
```

**Comment by FxKu** [comment]
:+1:

**Review by CyberDem0n — APPROVED** [review]

**Comment by CyberDem0n** [comment]
:+1:

**Comment by CyberDem0n** [comment]
Merged, thank you @ssinger!

**Pull request merged**